### ISSUE TITLE
Android app language resets to system default on its own

### ISSUE THREAD

**Issue by equeim** [organicmaps/organicmaps#8734]
After changing the per-app language in Android settings (feature that appeared in Android 13), the language is automatically reset to system default after ~~a few days~~ reboot.

**Steps to reproduce**
1. Go to Organic Maps' App Info page in system settings
2. Click on Language
3. Select language other than system default
4. ~~Wait for a few days~~ Reboot the device
5. Observe that Organic Maps' language was reset to system default

**Expected behaviour**
Language selected by the user remains in use.

**System information:**
 - Operating system and its version: Android 14
 - Organic Maps version: 2024.07.08-3-Google
 - Device Model: Samsung Galaxy S23

**Comment by Jean-BaptisteC** [comment]
We can't do anything on this, you can create a ticket at the support Google :)

**Comment by biodranik** [comment]
@equeim I've never observed Samsung resetting the language. Is it easily reproducible for any selected language or only for some specific language?

**Comment by equeim** [comment]
> @equeim I've never observed Samsung resetting the language. Is it easily reproducible for any selected language or only for some specific language?

I just did some experimenting and apparently it's triggered by reboot (I have set up my phone to automatically reboot once a week). Every time the phone is rebooted, the language on App info screen is reset to system default. This happens for any language, and only for Organic Maps.

**Comment by biodranik** [comment]
An interesting issue, likely a bug in the Samsung code. Let's keep it open, maybe someone will confirm it or suggest a workaround.

**Comment by equeim** [comment]
@biodranik I can reproduce it on Android 15 emulator, as well as real device with LineageOS 21 (Android 14).

**Comment by DMyachin** [comment]
Confirm. All my Pixel devices: 6-9. Stock and GrapheneOS

**Comment by biodranik** [comment]
It is definitely a system issue, we can't fix it in OM. What can help is finding an existing Google ticket (or opening a new one), and upvoting it.

**Comment by DMyachin** [comment]
I have 10+ apps with locale that was set with feature. I.e. `en_US` is default locale on my devices but for many apps I set `ru`. If I know an app created by russian speakers I use `ru` for the app.

Only Organic has the issue, sorry.

**Comment by biodranik** [comment]
You are right, I was able to reproduce the issue on the Android 15 device. Weird, we're following recommendations from https://developer.android.com/guide/topics/resources/app-languages
